repositories/db_executor.go

```diff
@@ -9,30 +9,69 @@ import (
 	"github.com/jackc/pgx/v5/pgconn"
 )
 
-type transactionOrPool interface {
+// //////////////////////////////////
+// Generic db executor (tx or pool)
+// //////////////////////////////////
+type TransactionOrPool interface {
 	Exec(ctx context.Context, sql string, arguments ...any) (pgconn.CommandTag, error)
 	Query(ctx context.Context, sql string, args ...any) (pgx.Rows, error)
 	QueryRow(ctx context.Context, sql string, args ...any) pgx.Row
 }
 
-// implements the DbExecutor from usecasess
-type ExecutorPostgres struct {
-	databaseShema models.DatabaseSchema
-	exec          transactionOrPool
+type PgExecutor struct {
+	databaseSchema models.DatabaseSchema
+	exec           TransactionOrPool
 }
 
-func (e ExecutorPostgres) DatabaseSchema() models.DatabaseSchema {
-	return e.databaseShema
+func (e PgExecutor) DatabaseSchema() models.DatabaseSchema {
+	return e.databaseSchema
 }
 
-func (e ExecutorPostgres) Exec(ctx context.Context, sql string, args ...any) (pgconn.CommandTag, error) {
+func (e PgExecutor) Exec(ctx context.Context, sql string, args ...any) (pgconn.CommandTag, error) {
 	return e.exec.Exec(ctx, sql, args...)
 }
 
-func (e ExecutorPostgres) Query(ctx context.Context, sql string, args ...any) (pgx.Rows, error) {
+func (e PgExecutor) Query(ctx context.Context, sql string, args ...any) (pgx.Rows, error) {
 	return e.exec.Query(ctx, sql, args...)
 }
 
-func (e ExecutorPostgres) QueryRow(ctx context.Context, sql string, args ...any) pgx.Row {
+func (e PgExecutor) QueryRow(ctx context.Context, sql string, args ...any) pgx.Row {
 	return e.exec.QueryRow(ctx, sql, args...)
 }
+
+////////////////////////////////////
+// Transaction
+////////////////////////////////////
+
+type PgTx struct {
```

Comment:
we could, as an extension and when we need it, make this PgTx struct (and the interface it implements in `repositories/db_executor_getter.go`) also expose the other methods from pgx.Tx (or something slighly more adapted to our use case), so that we could for instance manually create checkpoints, rollback to checkpoints, etc.
But for now just having the additional `RawTx()` method allows to distinguish the interfaces.